vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -199,21 +205,41 @@ def __init__(
             params_dtype=params_dtype,
             weight_loader=self.weight_loader)
 
-    def weight_loader(self, param: torch.nn.Parameter,
-                      loaded_weight: torch.Tensor, weight_name: str,
-                      shard_id: str, expert_id: int) -> None:
-        if shard_id not in ("w1", "w2", "w3"):
-            raise ValueError(f"shard_id must be ['w1','w2','w3'] but "
-                             f"got {shard_id}.")
-
-        # Special case for fp8 scales.
-        if getattr(param, "is_fp8_scale", False):
-            self._load_fp8_scale(param.data, loaded_weight, weight_name,
-                                 shard_id, expert_id)
-            return
+    def _load_per_tensor_weight_scale(self, shard_id: str,
```

Comment:
could these be utilities that can be shared with autogptq and autoawq?

Reply:
Yep